modules/admaticBidAdapter.js

```diff
@@ -1,117 +1,158 @@
-  supportedMediaTypes: ['video', 'banner'],
-  /**
-   * Determines whether or not the given bid request is valid.
-   *
-   * @param {BidRequest} bid The bid params to validate.
-   * @return boolean True if this is a valid bid, and false otherwise.
+  code: BIDDER_CODE,
+  aliases: [
+    {code: 'pixad'}
+  ],
+  supportedMediaTypes: [BANNER, VIDEO],
+  /** f
+   * @param {object} bid
+   * @return {boolean}
    */
-  isBidRequestValid: function(bid) {
+  isBidRequestValid: (bid) => {
     let isValid = false;
-    if (typeof bid.params !== 'undefined') {
-      let isValidNetworkId = _validateId(getValue(bid.params, 'networkId'));
-      isValid = isValidNetworkId;// && isValidTypeId;
+    if (bid?.params) {
+      const isValidNetworkId = _validateId(getValue(bid.params, 'networkId'));
+      const isValidHost = _validateString(getValue(bid.params, 'host'));
+      isValid = isValidNetworkId && isValidHost;
     }
 
     if (!isValid) {
-      logError('AdMatic networkId parameters are required. Bid aborted.');
+      logError(`${bid.bidder} networkId and host parameters are required. Bid aborted.`);
     }
     return isValid;
   },
+
   /**
-   * Make a server request from the list of BidRequests.
-   *
-   * @param {validBidRequests[]} an array of bids
-   * @return ServerRequest Info describing the request to the server.
+   * @param {BidRequest[]} validBidRequests
+   * @return {ServerRequest}
    */
-  buildRequests: function(validBidRequests, bidderRequest) {
+  buildRequests: (validBidRequests, bidderRequest) => {
     const bids = validBidRequests.map(buildRequestObject);
     const networkId = getValue(validBidRequests[0].params, 'networkId');
+    const host = getValue(validBidRequests[0].params, 'host');
     const currency = getValue(validBidRequests[0].params, 'currency') || 'TRY';
-
-    setTimeout(() => {
-      loadExternalScript(SYNC_URL, BIDDER_CODE);
-    }, bidderRequest.timeout);
+    const bidderName = validBidRequests[0].bidder;
 
     const payload = {
-      'user': {
-        'ua': navigator.userAgent
+      user: {
+        ua: navigator.userAgent
       },
-      'blacklist': [],
-      'site': {
-        'page': location.href,
-        'ref': location.origin,
-        'publisher': {
-          'name': location.hostname,
-          'publisherId': networkId
+      blacklist: [],
+      site: {
+        page: location.href,
+        ref: location.origin,
+        publisher: {
+          name: location.hostname,
+          publisherId: networkId
         }
       },
       imp: bids,
       ext: {
-        'cur': currency,
-        'type': 'admatic'
+        cur: currency,
+        bidder: bidderName
       }
     };
 
-    const payloadString = JSON.stringify(payload);
-    return {
-      method: 'POST',
-      url: ENDPOINT_URL,
-      data: payloadString,
-      options: {
-        contentType: 'application/json'
-      }
-    };
+    setTimeout(() => {
+      loadExternalScript(SYNC_URL, bidderName);
```

Comment:
UserSync *must* be done in `getUserSyncs` function: 
https://docs.prebid.org/dev-docs/bidder-adaptor.html#registering-user-syncs

Reply:
Hi @musikele ,
thank you for feedback.
Development done. 👍